object/datastore.go

```diff
@@ -105,25 +105,29 @@ func (d Datastore) Path(path string) string {
 	}).String()
 }
 
-// NewURL constructs a url.URL with the given file path for datastore access over HTTP.
-func (d Datastore) NewURL(path string) *url.URL {
-	u := d.c.URL()
-
-	scheme := u.Scheme
+// NewDatastoreURL constructs a url.URL with the given file path for datastore access over HTTP.
+func NewDatastoreURL(base url.URL, dcPath, dsName, path string) *url.URL {
+	scheme := base.Scheme
 	// In rare cases where vCenter and ESX are accessed using different schemes.
 	if overrideScheme := os.Getenv("GOVMOMI_DATASTORE_ACCESS_SCHEME"); overrideScheme != "" {
 		scheme = overrideScheme
 	}
 
-	return &url.URL{
-		Scheme: scheme,
-		Host:   u.Host,
-		Path:   fmt.Sprintf("/folder/%s", path),
-		RawQuery: url.Values{
-			"dcPath": []string{d.DatacenterPath},
-			"dsName": []string{d.Name()},
-		}.Encode(),
-	}
+	base.Scheme = scheme
+	base.Path = fmt.Sprintf("/folder/%s", path)
+	base.RawQuery = url.Values{
+		"dcPath": []string{dcPath},
+		"dsName": []string{dsName},
+	}.Encode()
+
+	return &base
+}
+
+// NewURL constructs a url.URL with the given file path for datastore access over HTTP.
```

Comment:
Maybe add a comment here that the caller needs to call "FindInventoryPath" (or some other method) that sets the datacenterPath in the DS object?

Reply:
Thanks @aruneshpa , agree, added.